### DISCUSSION TITLE
API for graphs

### DISCUSSION THREAD

**Discussion by marcelobalbinot** [akvorado/akvorado#1276] [Q&A]
I believe that an API where parameters can be sent and the graph returned in a .png image would make a lot of sense for many users of the tool.

I tried to understand if this is already possible, and it seems that it is not!

Are there any plans for a similar functionality?

**Comment by vincentbernat** [comment]
There is a plan for an integration into Grafana. I know some are already doing that but the JSON API is currently unstable and not documented. The plan is to have a versioned API for graphs and a very simple Grafana plugin.

**Reply by marcelobalbinot** [reply]
Sure, thanks for the answer!
I'm following this same topic, because it's related...
Currently, the URL with parameters is encoded. If we could assemble the URL in plain text, or if we knew how to encode the parameters, I believe that this would already give the user some ways to integrate with other platforms.

Is this encoding/decoding of the URL public? Or, could it be public?

**Reply by vincentbernat** [reply]
The URL is not really the API. Open the devtools and see the call to /api/v0/graph. That's the API. But as hinted by the v0, it's not versioned and internal.
